### PR TITLE
Update to 2.4.2 and improve build script

### PR DESCRIPTION
**Pull request by diondokter**
Apparently installing `llvm-tools-preview` doesn't register the binaries on path.
So I think this is the way to do it.

CI wasn't happy about the old way.
So I'd like to do a small release for this

### CODE REVIEW

**Comment by jonathanpallant** [comment]
1. I think the version number is supposed to match the underlying Nordic C nrfxlib version, so this would be 2.1.0-r1 or something.

2. Also the rustup component is now `rustup component add llvm-tools`

3. `cargo-binutils` does it like this: https://github.com/rust-embedded/cargo-binutils/blob/master/src/rustc.rs#L15

**Comment by diondokter** [comment]
Thanks for the review @jonathanpallant !

1. That's what been done yes, and I think we should broadly keep doing it. But I think your proposed version is only meant for prereleases. And as such, cargo will think it's incompatible and even precedes the current 2.1.0. By making it 2.1.1 the number stays roughly the same, but is counted as a patch.
(There was a 2.1.1 release of nrfxlib, which makes this all a bit meh)
What do you think? Prerelease tag and incompatible with the old version or patch tag and a bit confusing?
2. Ah, I thought that was still in nightly. I'll change the docs.
3. The used llvm-tools crate does it in broadly the same way, calling rustc for the sysroot. It's a crate by phil-opp. I think it's better to use the crate, than copy the implementation from binutils

**Comment by jonathanpallant** [comment]
1. Ugh, yeah - https://doc.rust-lang.org/cargo/reference/resolver.html#pre-releases. You can't use `2.0.0+r1` [either](https://doc.rust-lang.org/cargo/reference/resolver.html#version-metadata). It'll cause confusion if the versions don't match, so I think you'll have to bump nrfxlib at the same time?
2. Cool
3. Fair enough.

**Comment by diondokter** [comment]
If I'm gonna spend the time to update to a newer version, then I want to do it well.
It looks like going to 2.4.1 isn't too bad.

**Comment by diondokter** [comment]
Ok, that was relatively easy and I think I got every relevant new header too

**Comment by diondokter** [comment]
While I'm at it, shall I configure a simple github CI and remove the travis CI?

**Comment by jonathanpallant** [comment]
In for a penny, in for a pound!

(or, yes, please do fix the CI)

**Comment by mfiumara** [comment]
Could this be updated to 2.4.2 and merged? https://github.com/nrfconnect/sdk-nrf/releases/tag/v2.4.2

**Comment by diondokter** [comment]
> Could this be updated to 2.4.2 and merged? https://github.com/nrfconnect/sdk-nrf/releases/tag/v2.4.2

You're in luck :)
A colleague of mine has planned to pick this up this week :)

**Comment by diondokter** [comment]
Ok, this has been updated to 2.4.2 by @tdittr and it's working with the nrf-modem crate

File: README.md [hunk 1]
```diff
@@ -65,9 +65,14 @@ without any additional terms or conditions.
 
 ## Changelog
 
-### Unreleased Changes ([Source](https://github.com/nrf-rs/nrfxlib-sys/tree/master) | [Changes](https://github.com/nrf-rs/nrfxlib-sys/compare/v2.1.0...develop))
+### Unreleased Changes ([Source](https://github.com/nrf-rs/nrfxlib-sys/tree/develop) | [Changes](https://github.com/nrf-rs/nrfxlib-sys/compare/v2.4.1...develop))
```
Comment (jonathanpallant) [inline README.md]:
```suggestion
### Unreleased Changes ([Source](https://github.com/nrf-rs/nrfxlib-sys/tree/develop) | [Changes](https://github.com/nrf-rs/nrfxlib-sys/compare/v2.4.2...develop))
```

**Review by jonathanpallant — APPROVED** [review]
LGTM - just one typo.

**Comment by diondokter** [comment]
Thanks for the review!

**Comment by diondokter** [comment]
@tdittr and @mfiumara It's now published to crates.io :)

**Comment by mfiumara** [comment]
Nice thanks for the quick response :)

**Pull request merged**